src/config/paths.ts

```diff
@@ -1,9 +1,20 @@
+import { ItemLayoutMode } from '@/enums';
+
 export const HOME_PATH = '/';
 export const BOOKMARKED_ITEMS_PATH = '/bookmarks';
 export const PUBLISHED_ITEMS_PATH = '/published';
 export const SIGN_UP_PATH = '/signUp';
 export const ITEMS_PATH = '/items';
-export const buildItemPath = (id = ':itemId'): string => `${ITEMS_PATH}/${id}`;
+export const buildItemPath = (
+  id = ':itemId',
+  args: { mode?: ItemLayoutMode } = {},
+): string => {
+  let qs = '';
+  if (args.mode) {
+    qs = `?mode=${args.mode}`;
+  }
```

Comment:
I think you can use `URLSearchParams` to build it:
```suggestion
  const search = new URLSearchParams();
  if (args.mode) {
    search.set("mode", args.mode);
  }
  // use then search.toString()
```